docs/workflows/common-workflow-steps/custom-script.md

```diff
@@ -510,4 +511,112 @@ echo "Completed Build Log URL: $AC_COMPLETED_BUILD_LOG_URL"
 # To use these URL variables in later steps within the same workflow, add them to ENV_FILE_PATH as shown below; otherwise, they will not be accessible.
 echo "AC_COMPLETED_BUILD_LOG_URL=$AC_COMPLETED_BUILD_LOG_URL" >> $AC_ENV_FILE_PATH
 echo "AC_IN_PROGRESS_BUILD_LOG_URL=$AC_IN_PROGRESS_BUILD_LOG_URL" >> $AC_ENV_FILE_PATH
-```
\ No newline at end of file
+```
+
+### How do I store and re-use scripts from a Git repository for Custom Script steps?
+
+If you are looking for a modular way to manage your Appcircle CI custom scripts, you can host your script files in a Git repository and pull them dynamically into the build environment. This approach helps keep your workflow clean and allows centralized version control for your custom scripts.
+
+This section explains how to clone a private Git repository and execute a specific script file from it using the Custom Script step.
+
+:::info
+
+If you are using public repository, you don't need to obtain an access token. In this case, use the Git URL without the authentication command in the script below.
+
+:::
+
+To use this approach, you must first convert your Bash or Ruby script into a Git repository. This means placing your Bash or Ruby file into a Git repository and pushing it to your preferred Git provider (e.g., GitHub, GitLab, Bitbucket, Azure).
+
+:::caution
+
+It is recommended to create a repository specifically for the desired custom script as a Bash or Ruby file.
+
+:::
+
+:::info Required Permissions
+
+To successfully clone a repository, your personal access token (PAT) must have at least read access to the repository.
+
+Write or admin permissions are not required for this use case.
+
+Note: Bitbucket requires a user-level personal access token for Git operations such as cloning. Repository-level access tokens are not supported for Git over HTTPS and will result in authentication errors if used.
+
+:::
+
+<PatDanger />
+
+```bash
+
+#Bash script
+set -e
+
+# Repository details
+CS_REPO_CLONE_URL="https://example.org/exampleuser/examplerepo.git"
+CS_USERNAME="$YOUR_USERNAME"
+CS_PAT="$YOUR_PAT"
+CS_GIT_BRANCH="main" # or desired branch name 
+CS_SCRIPT_FILE="test.sh"  # or "Example.rb"
+
+# Create Authorization header using Base64 encoding
+AUTH_STRING="$CS_USERNAME:$CS_PAT"
+ENCODED_AUTH=$(printf "%s" "$AUTH_STRING" | base64)
+HEADER_VALUE="Authorization: Basic $ENCODED_AUTH"
+
+# Clone the repository using the Authorization header
```

Comment:
Before proceeding to the Git Clone command, please create a new directory (with an appropriate name) and perform the `git clone` operation inside that directory.
Avoiding Git operations directly in the root directory is a safer approach.
It would be preferable to use a unique name for this directory.